music_metadata_extractor/songdata.py

```diff
@@ -10,9 +10,14 @@ class SongData:
     def __init__(self, input_url: str):
         """Constructor for the SongData class."""
         provider_input: BaseProviderInput = get_provider_input(input_url)
-        provider_data: BaseProvider = SpotifyProvider(provider_input).data
-        self.track: Track = provider_data.track
-        self.artists: List[Artist] = provider_data.artists
+        try:
+            provider_data: BaseProvider = SpotifyProvider(provider_input).data
+            self.track: Track = provider_data.track
+            self.artists: List[Artist] = provider_data.artists
+        except:
```

Comment:
Can you specifically catch IndexError rather than all exceptions, so we don't miss out on any unintended errors in future?

Also, please add a warning log (using the Python logging module) with an error message and the stack trace.

Reply:
Ok I'll look into it and update by tomorrow.